test/test_table.py

```diff
@@ -0,0 +1,147 @@
+
+    def test_setitem(self):
+        tbl = HeaderTable()
+        with pytest.raises(TypeError) as einfo:
+            tbl[1] = (b'TestName', b'TestValue')
+        assert 'HeaderTable' in str(einfo.value)
+
+    def test_repr(self):
+        tbl = HeaderTable()
+        tbl.add(b'TestName1', b'TestValue1')
+        tbl.add(b'TestName2', b'TestValue2')
+        tbl.add(b'TestName2', b'TestValue2')
+        # Meh, I hate that I have to do this to test
+        # repr
+        if(is_py3):
+            exp = ("HeaderTable(4096, False, ["    +
+                   "(b'TestName2', b'TestValue2'), " +
+                   "(b'TestName2', b'TestValue2'), " +
+                   "(b'TestName1', b'TestValue1')"   +
+                   "])")
+        else:
+            exp = ("HeaderTable(4096, False, ["    +
+                   "('TestName2', 'TestValue2'), " +
+                   "('TestName2', 'TestValue2'), " +
+                   "('TestName1', 'TestValue1')"   +
+                   "])")
+        res = str(tbl)
+        assert res == exp
+
+    def test_add_to_large(self):
+        tbl = HeaderTable()
+        # Max size to small to hold the value we specify
+        tbl.maxsize = 17 + 31
+        tbl.add(b'TestName', b'TestValue')
+        # Table length should be 0
+        assert len(tbl) == 0
+
+    def test_search_in_static_full(self):
+        tbl = HeaderTable()
+        itm = HeaderTable.STATIC_TABLE[0]
+        exp = (1, itm[0], itm[1])
+        res = tbl.search(itm[0], itm[1])
+        assert res == exp
+
+    def test_search_in_static_partial(self):
+        tbl = HeaderTable()
+        itm = HeaderTable.STATIC_TABLE[0]
+        exp = (1, itm[0], None)
+        res = tbl.search(itm[0], b'NotInTable')
+        assert res == exp
+
+    def test_search_in_dynamic_full(self):
+        tbl = HeaderTable()
+        idx = len(HeaderTable.STATIC_TABLE)+1
+        tbl.add(b'TestName', b'TestValue')
+        exp = (idx , b'TestName', b'TestValue')
+        res = tbl.search(b'TestName', b'TestValue')
+        assert res == exp
+
+    def test_search_in_dynamic_partial(self):
+        tbl = HeaderTable()
+        idx = len(HeaderTable.STATIC_TABLE)+1
+        tbl.add(b'TestName', b'TestValue')
+        exp = (idx , b'TestName', None)
+        res = tbl.search(b'TestName', b'NotInTable')
+        assert res == exp
+
+    def test_search_no_match(self):
+        tbl = HeaderTable()
+        idx = len(HeaderTable.STATIC_TABLE)
+        tbl.add(b'TestName', b'TestValue')
+        exp = None
+        res = tbl.search(b'NotInTable', b'NotInTable')
+        assert res == exp
+
+    def test_maxsize_prop_getter(self):
+        tbl = HeaderTable()
+        assert tbl._maxsize == HeaderTable.DEFAULT_SIZE
+
+    def test_maxsize_prop_setter(self):
+        tbl = HeaderTable()
+        exp = int(HeaderTable.DEFAULT_SIZE / 2)
+        tbl.maxsize = exp
+        assert tbl.resized == True
+        assert tbl._maxsize == exp
```

Comment:
We don't really care if the backing variable changed, we just care that the observable contract is obeyed.